### PR TITLE
Add ReadmeFileUrl field to IPackageSearchMetadata

### PR DESCRIPTION
**Pull request by jgonz120**
# Bug

Part of: https://github.com/NuGet/Home/issues/12583

## Description
Adds the ReadmeFileUrl field to IPackageSearchMetadata and implements the ability to get the URL for local packages. Future PRs will include implementations for getting the URL for remote READMEs and downloading them.  

## PR Checklist

- [X] Meaningful title, helpful description and a linked NuGet/Home issue
- [X] ~~Added tests~~
- [X] ~~Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.~~

### CODE REVIEW

File: test/TestUtilities/Test.Utility/Protocol/V3PackageSearchMetadataFixture.cs [hunk 1]
```diff
@@ -74,6 +74,8 @@ public MockPackageSearchMetadata()
 
             public Uri ReadmeUrl => null;
 
+            public Uri ReadmeFileUrl => null;
```
Comment (donnie-msft) [inline test/TestUtilities/Test.Utility/Protocol/V3PackageSearchMetadataFixture.cs]: I was thinking there should be a few more tests that need this new property added to existing tests.

Some new tests may be needed as well, like this Local metadata for IconUri, for example: checked:https://github.com/NuGet/NuGet.Client/blob/e2f077a9a062e8190b3482e29786a758182b33fb/test/NuGet.Core.Tests/NuGet.Protocol.Tests/LocalPackageSearchMetadataTests.cs#L62

Comment (jgonz120) [reply test/TestUtilities/Test.Utility/Protocol/V3PackageSearchMetadataFixture.cs]: I was planning on adding the unit tests as part of another PR, do you think I should do it now?

Comment (jgonz120) [reply test/TestUtilities/Test.Utility/Protocol/V3PackageSearchMetadataFixture.cs]: Meh, added a unit test to match the one you linked. It looks like there's only one test class for LocalPackageSearchMetadata.

**Review by donnie-msft — APPROVED** [review]

File: src/NuGet.Clients/NuGet.VisualStudio.Internal.Contracts/TransitivePackageSearchMetadata.cs [hunk 2]
```diff
@@ -59,6 +61,7 @@ public class TransitivePackageSearchMetadata : IPackageSearchMetadata
 
         public IEnumerable<PackageVulnerabilityMetadata> Vulnerabilities => _packageSearchMetadata.Vulnerabilities;
 
+
```
Comment (zivkan) [inline src/NuGet.Clients/NuGet.VisualStudio.Internal.Contracts/TransitivePackageSearchMetadata.cs]: nitpick: unnecessary whitespace line

File: src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadata.cs [hunk 3]
```diff
@@ -106,6 +106,9 @@ public string Owners
         [JsonConverter(typeof(SafeUriConverter))]
         public Uri ReadmeUrl { get; private set; }
 
+        [JsonIgnore]
+        public Uri ReadmeFileUrl { get; private set; }
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadata.cs]: Why ignore this from JSON?   Don't we want HTTP sources to be able to provide the raw markdown file?

The PR description also doesn't explain.

File: src/NuGet.Core/NuGet.Protocol/Model/IPackageSearchMetadata.cs [hunk 4]
```diff
@@ -24,6 +24,7 @@ public interface IPackageSearchMetadata
         Uri LicenseUrl { get; }
         Uri ProjectUrl { get; }
         Uri ReadmeUrl { get; }
+        Uri ReadmeFileUrl { get; }
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.Protocol/Model/IPackageSearchMetadata.cs]:
```suggestion
        string ReadmeFileUrl { get; }
```

It's a perf mistake to use `System.Uri`.  It's expensive to calculate, and `HttpClient` has overloads that allow you to pass the URL as a string. Maybe it parses as an Uri internally, I don't know, but it also doesn't matter. This property isn't always going to be used, so we shouldn't pay the perf cost to force it to always be parsed.

I'd love to change the other ones from `Uri` to `string`, but that would be a breaking change in a package we ship. But we can avoid making the same mistake in new APIs.

File: src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadata.cs (hunk 3 above)
Comment (jgonz120) [reply src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadata.cs]: I wanted to focus on the local package reading for this PR. In my head I've been keeping them separate, then in a future PR once all the remote reading stuff is added I was going to update this field.

**Review by Nigusu-Allehu — APPROVED** [review]

File: src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadata.cs (hunk 3 above)
Comment (jgonz120) [reply src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadata.cs]: Updated~
